### ISSUE TITLE
Add OpenAL Soft library for Android

### ISSUE THREAD

**Issue by roeyskoe** [dotnet/Silk.NET#617]
# Summary of feature
Currently OpenAL Soft library is not compiled for Android, and trying to use it throws an `FileNotFoundException`.

# Does this have a proposal?
No

**Comment by Perksey** [comment]
Will only be adding this for .NET 6 as backporting to Xamarin would be a lot of maintenance burden.

**Comment by Perksey** [comment]
Low priority, won't make it into 2.9.

**Comment by Pyrdacor** [comment]
Ok with a valid RID in the csproj, the exception is gone. It just won't find the .so file now. :)

**Comment by Perksey** [comment]
Current progress:
- @roeyskoe did some work to standardize the build process for native libraries
- To my knowledge, Android does not respect the runtimes folder.
- As such, we'll need someone (community) to create a gradle project for producing an `aar` for this.

We have the SilkDroid gradle project to go off, but beyond that this is a large task full of a lot of figuring out.

The Silk.NET team at this time don't have any real priority on this issue due to low demand and time constraints thus it is a community issue (help wanted!)

**Comment by Pyrdacor** [comment]
Ok thanks for the heads-up. Maybe this will help: https://github.com/sgothel/openal-soft/blob/master/Android.md

Can't tell if I will have the time for it either.

**Comment by roeyskoe** [comment]
I already have gotten OpenAL to build for Android, but have not yet have time to research how the required(?) `aar` could be done.

I could actually open a pr for the OpenAL native builds, even without Android support.

**Comment by Pyrdacor** [comment]
Nevermined I just used the AudioTrack class for Android which works quite well. :)

**Comment by alexrp** [comment]
Once #2049 is merged, we'll be using CMake toolchain files for cross-compilation. That *should* make it a lot easier to use the Android NDK to cross-compile OpenAL Soft libraries for Android if anyone is still interested in this. I probably won't do it myself, but can offer assistance to anyone willing to drive the effort.

**Comment by orosbogdan** [comment]
> Once #2049 is merged, we'll be using CMake toolchain files for cross-compilation. That _should_ make it a lot easier to use the Android NDK to cross-compile OpenAL Soft libraries for Android if anyone is still interested in this. I probably won't do it myself, but can offer assistance to anyone willing to drive the effort.

seems this is merged, did this fix openal soft on android ?

**Comment by Perksey** [comment]
Not yet - please see the latter half of the comment you quoted.
